language/compiler/ir_to_bytecode/syntax/src/syntax.rs

```diff
@@ -1 +1,1798 @@
-include!(concat!(env!("OUT_DIR"), "/syntax.rs"));
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+use std::fmt;
+use codespan::{ByteIndex, Span};
+
+use crate::ast::{
+    ModuleDefinition, StructDefinition, Script, Program, ScriptOrModule,
+    FunctionBody, FunctionVisibility, ImportDefinition, ModuleName,
+    Block, Block_, Cmd, CopyableVal, CopyableVal_, Spanned, Kind, TypeVar,
+    Exp_, Exp, Var,  Var_, FunctionCall, FunctionCall_,
+    FunctionName, Builtin, Statement, IfElse, While, Loop, Type, Fields,
+    StructName, QualifiedStructIdent, Function, BinOp, ModuleIdent, QualifiedModuleIdent, UnaryOp,
+    LValue, LValue_, parse_field, Field_, TypeVar_, Function_, StructDefinition_,
+};
+use crate::lexer::*;
+use libra_types::{account_address::AccountAddress, byte_array::ByteArray};
+use hex;
+
+// FIXME: The following simplified version of ParseError copied from
+// lalrpop-util should be replaced.
+
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum ParseError<L, T, E> {
+    InvalidToken { location: L },
+    UnrecognizedToken {
+        token: (L, T, L),
+        expected: Vec<String>,
+    },
+    User { error: E },
+}
+
+/// Format a list of expected tokens.
+fn fmt_expected(f: &mut fmt::Formatter, expected: &[String]) -> fmt::Result {
+    if !expected.is_empty() {
+        writeln!(f)?;
+        for (i, e) in expected.iter().enumerate() {
+            let sep = match i {
```

Comment:
Nit: might be slightly easier to read as an `if/else if/else`

Reply:
I'm not even sure what this code is about. I copied it from lalrpop-util. Per the preceding "FIXME" comment, I think we should just replace all of this code from lalrpop instead of trying to fix it up.